### ISSUE TITLE
Add tab player sorting and prefixes to the API

### ISSUE THREAD

**Issue by SlenkyDev** [PaperMC/Velocity#605]
Hi there,
I would like to request a new feature within the Velocity plugin API that has been bothering me for a while.

Add tab player sorting and prefixes to the API..

This prevents me from using a proxy side tab plugin which makes me unable to use stuff like server name prefixes ext. in the tab list without manually specifying so.

https://github.com/NEZNAMY/TAB/issues/390#issuecomment-924087165

**Comment by astei** [comment]
This is really a request for a scoreboard API in Velocity. It's something that has been in the works for a while but I have no news to share.

**Comment by purejosh** [comment]
Yes, scoreboard API would be nice to see in new Velocity builds. 👍

**Comment by SlenkyDev** [comment]
> This is really a request for a scoreboard API in Velocity. It's something that has been in the works for a while but I have no news to share.

Anything new? I hate having to edit all of my servers for one simple change.

**Comment by Goofert42** [comment]
Is this feature planned or no? I want a global playerlist for my network but also want prefix's and player sorting. It doesn't seem to work without me putting TAB on each induvidial server, sacrificing the global player list.

**Comment by Timongcraft** [comment]
Well in modern versions you can directly set the order, see: https://jd.papermc.io/velocity/3.4.0/com/velocitypowered/api/proxy/player/TabListEntry.html#setListOrder(int)
and setting the tab name has long been possible, see: https://jd.papermc.io/velocity/3.4.0/com/velocitypowered/api/proxy/player/TabListEntry.html#setDisplayName(net.kyori.adventure.text.Component)

**Comment by Goofert42** [comment]
> Well in modern versions you can directly set the order, see: https://jd.papermc.io/velocity/3.4.0/com/velocitypowered/api/proxy/player/TabListEntry.html#setListOrder(int)
> and setting the tab name has long been possible, see: https://jd.papermc.io/velocity/3.4.0/com/velocitypowered/api/proxy/player/TabListEntry.html#setDisplayName(net.kyori.adventure.text.Component)

So why doesn't TAB show the prefixes or sort players based on their permission group?

**Comment by Timongcraft** [comment]
How would I know? You'd need to ask them

**Comment by purejosh** [comment]
> So why doesn't TAB show the prefixes or sort players based on their permission group?

Because Scoreboard API hasn't been implemented in velocity, yet. I've just accepted that sorting by perm groups won't be a thing. Prefixes, however, works on my server network.

**Comment by Timongcraft** [comment]
> Because Scoreboard API hasn't been implemented in velocity, yet. I've just accepted that sorting by perm groups won't be a thing. Prefixes, however, works on my server network.

Velocity doesn't need a Scoreboard API for that to work though. It has no concept of 'permission groups' or any sort of grouping for that matter. So you would have to depend or a 3rd party to provide that or do it yourself, but a full-blown Scoreboard API is not needed for tablist sorting and formatting.

**Comment by Goofert42** [comment]
> > So why doesn't TAB show the prefixes or sort players based on their permission group?
> 
> Because Scoreboard API hasn't been implemented in velocity, yet. I've just accepted that sorting by perm groups won't be a thing. Prefixes, however, works on my server network.

I have a plugin installed on my proxy that adds support for scoreboards. Player prefixes and sorting shouldn't need the scoreboard API.

**Comment by electronicboy** [comment]
Player prefixes are 100% controlled by the scoreboard, not tab, and would require support for scoreboards; sorting is already implemented for the newer sorting mechanism in the display order field in the tab, anything else sorting wise relies on scoreboard team name sorting (backed by name sorting)
